apps/permissions.go

```diff
@@ -47,12 +43,10 @@ func (p Permissions) Contains(permission Permission) bool {
 func (p Permission) String() string {
 	m := ""
 	switch p {
-	case PermissionUserJoinedChannelNotification:
-		m = "be notified when users join channels"
 	case PermissionActAsUser:
-		m = "use Mattermost REST API as connected users"
+		m = "Use Mattermost REST API as the connected user"
```

Comment:
I know we are replacing here previous code, but I was wondering your opinion about a different approach when using these types of switches.

I'm thinking maybe my approach consumes more memory maybe? But In case we have 20 cases (for example), would be easier to maintain?

The idea would be:

```go
type Permission string
const (
	// PermissionActAsBot means that a Bot User will be created when the App is
	// installed. Call requests will then include the Bot access token, the app
	// can use them with the Mattermost REST API. The bot will not automatically
	// receive permissions to any resources, need to be added explicitly.
	PermissionActAsBot Permission = "act_as_bot"

	// PermissionActAsUser means that the app is allowed to connect users'
	// OAuth2 accounts, and then use user API tokens.
	PermissionActAsUser Permission = "act_as_user"

	// PermissionRemoteOAuth2 means that the app is allowed to use remote (3rd
	// party) OAuth2 support, and will store secrets to 3rd party system(s).
	PermissionRemoteOAuth2 Permission = "remote_oauth2"

	// PermissionRemoteWebhooks means that the app is allowed to receive webhooks from a remote (3rd
	// party) system, and process them as Bot.
	PermissionRemoteWebhooks Permission = "remote_webhooks"
)

...

func (p Permission) String() String {
    permissionMessages := map[Permission]string {
        PermissionActAsBot: "Use Mattermost REST API as the connected user",
        PermissionActAsUser: "Use Mattermost REST API as the app's bot user",
        PermissionRemoteOAuth2: "Use a remote (3rd party) OAuth2 and store secrets",
    }

    permission := PermissionActAsBot

    if _, ok := permissionMessages[permission]; ok {
        return permissionMessages[permission]
    }

    // default
   return "default message"
}
```

I have [a working example here](https://replit.com/@javaguirre/switch-case-scopes-example#main.go)

Thank you Lev!

Reply:
Ok! It makes sense, Thank you Lev! I really like the Scope approach better, much easier to test and more independent from the rest.